daq_macros.py

```diff
@@ -3117,20 +3131,46 @@ def zebraDaqBluesky(flyer, angle_start, scanWidth, imgWidth, exposurePeriodPerIm
       wavelength = beamline_lib.motorPosFromDescriptor("wavelength")
     det_distance_m = beamline_lib.motorPosFromDescriptor("detectorDist")
     if daq_utils.beamline == "nyx":
-        current_x *= 1000  # convert translation from mm to microns on NYX
-        current_y *= 1000
-        current_z *= 1000
-        det_distance_m /= 1000  # convert distance from mm to m on NYX
+      det_distance_m /= 1000  # convert distance from mm to m on NYX
     if daq_utils.beamline in ("nyx", "fmx"):
         transmission = getPvDesc("RI_Atten_SP")
     else:
         transmission = getPvDesc("transmissionRBV")
+    vectorParams["x_beam"]=x_beam
+    vectorParams["y_beam"]=y_beam
+    vectorParams["wavelength"]=wavelength
+    vectorParams["det_distance_m"]=det_distance_m
+    vectorParams["transmission"]=transmission
+    return vectorParams
+
+
+def zebraDaqBluesky(flyer, angle_start, num_images, scanWidth, imgWidth, exposurePeriodPerImage, filePrefix, data_directory_name, file_number_start, vector_params, scanEncoder=3, changeState=True):
+
+    logger.info("in Zebra Daq Bluesky #1")
+    logger.info(f" with vector: {vector_params}")
+    gov_lib.setGovRobot(gov_robot, "DA")
+
+    x_vec_start=vector_params["vecStart"]["x"]
+    y_vec_start=vector_params["vecStart"]["y"]
+    z_vec_start=vector_params["vecStart"]["z"]
+    x_vec_end=vector_params["vecEnd"]["x"]
+    y_vec_end=vector_params["vecEnd"]["y"]
+    z_vec_end=vector_params["vecEnd"]["z"]
+    if beamline == "nyx":
+      x_vec_start *= 1000
+      y_vec_start *= 1000
+      z_vec_start *= 1000
+      x_vec_end *= 1000
+      y_vec_end *= 1000
+      z_vec_end *= 1000
 
     flyer.update_parameters(angle_start=angle_start, scan_width=scanWidth, img_width=imgWidth, num_images=num_images, exposure_period_per_image=exposurePeriodPerImage, \
-                   x_start_um=current_x, y_start_um=current_y, z_start_um=current_z, \
+                   x_start_um=x_vec_start, y_start_um=y_vec_start, z_start_um=z_vec_start, \
+                   x_end_um=x_vec_end, y_end_um=y_vec_end, z_end_um=z_vec_end, \
                    file_prefix=filePrefix, data_directory_name=data_directory_name, file_number_start=file_number_start,\
-                   x_beam=x_beam, y_beam=y_beam, wavelength=wavelength, det_distance_m=det_distance_m,\
-                   detector_dead_time=0.024, scan_encoder=scanEncoder, change_state=changeState, transmission=transmission)
+                   x_beam=vector_params["x_beam"], y_beam=vector_params["y_beam"], wavelength=vector_params["wavelength"], det_distance_m=vector_params["det_distance_m"],\
+                   detector_dead_time=0.024, scan_encoder=scanEncoder, change_state=changeState, transmission=vector_params["transmission"])
```

Comment:
can you extract the dead time? this should be something like flyer.detector.dead_time.get() for Eiger detectors. that value is for Pilatus

Reply:
extracted dead time and added handling for attribute error in 1d73c26